core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -0,0 +1,241 @@
+      CommandLineUtils.printUsageAndDie(opts.parser, "Command must include exactly one action: --list, --add, --remove.")
+
+    val props = new Properties()
+    props.load(new FileInputStream(new File(opts.options.valueOf(opts.config))))
+
+    if (!props.containsKey(KafkaConfig.AuthorizerClassNameProp))
+      CommandLineUtils.printUsageAndDie(opts.parser, "Config file must set property = " + KafkaConfig.AuthorizerClassNameProp
+        + " to a class that implements Authorizer interface.")
+
+    val kafkaConfig = KafkaConfig.fromProps(props)
+    val authZ: Authorizer = CoreUtils.createObject(kafkaConfig.authorizerClassName)
+    authZ.configure(kafkaConfig.originals())
+
+    try {
+      if (opts.options.has(opts.addOpt))
+        addAcl(authZ, opts)
+      else if (opts.options.has(opts.removeOpt))
+        removeAcl(authZ, opts)
+      else if (opts.options.has(opts.listOpt))
+        listAcl(authZ, opts)
+    } catch {
+      case e: Throwable =>
+        println("Error while executing topic command " + e.getMessage)
+        println(Utils.stackTrace(e))
+    }
+  }
+
+  private def addAcl(authZ: Authorizer, opts: AclCommandOptions): Unit = {
+    val acls: Set[Acl] = getAcl(opts)
+
+    if (acls.isEmpty)
+      CommandLineUtils.printUsageAndDie(opts.parser, "You must specify one of : --allowprincipals, --denyPrincipals when trying to add acls.")
+
+    val resource = getResource(opts)
+
+    println("Adding acls: " + nl + acls.map("\t" + _).mkString(nl) + nl)
+    authZ.addAcls(acls, resource)
+
+    listAcl(authZ, opts)
+  }
+
+  private def removeAcl(authZ: Authorizer, opts: AclCommandOptions): Unit = {
+    val acls: Set[Acl] = getAcl(opts)
+    val resource = getResource(opts)
+    if (acls.isEmpty) {
+      if (confirmaAction("Are you sure you want to delete all acls for resource: " + getResource(opts) + " y/n?"))
+        authZ.removeAcls(resource)
+    } else {
+      if (confirmaAction(("Are you sure you want to remove acls: " + nl + acls.map("\t" + _).mkString(nl) + nl) + "  from resource " + getResource(opts) + " y/n?"))
+        authZ.removeAcls(acls, resource)
+    }
+
+    listAcl(authZ, opts)
+  }
+
+  private def listAcl(authZ: Authorizer, opts: AclCommandOptions): Unit = {
+    val acls: Set[Acl] = authZ.getAcls(getResource(opts))
+    println("Following is list of  acls for resource : " + getResource(opts) + nl + acls.map("\t" + _).mkString(nl) + nl)
+  }
+
+  private def getAcl(opts: AclCommandOptions): Set[Acl] = {
+    val allowedPrincipals = getPrincipals(opts, opts.allowPrincipalsOpt)
+
+    val deniedPrincipals = getPrincipals(opts, opts.denyPrincipalsOpt)
+
+    val allowedHosts = getHosts(opts, opts.allowHostsOpt, opts.allowPrincipalsOpt)
+
+    val deniedHosts = getHosts(opts, opts.denyHostssOpt, opts.denyPrincipalsOpt)
+
+    val operations = if (opts.options.has(opts.operationsOpt))
+      opts.options.valueOf(opts.operationsOpt).toString.split(delimeter).map(s => Operation.fromString(s)).toSet
```

Comment:
You can replace `.map(s => Operation.fromString(s))` with `.map(Operation.fromString)`